test/widgets/message_list_test.dart

```diff
@@ -922,6 +922,72 @@ void main() {
         await tester.pump();
         tester.widget(find.text('new stream name'));
       });
+
+      testWidgets('navigates to TopicNarrow on tapping topic in CombinedFeedNarrow', (tester) async {
+        final pushedRoutes = <Route<void>>[];
+        final navObserver = TestNavigatorObserver()
+          ..onPushed = (route, prevRoute) => pushedRoutes.add(route);
+
+        final channel = eg.stream();
+        final message = eg.streamMessage(stream: channel, topic: 'topic name');
+
+        await setupMessageListPage(
+          tester,
+          narrow: const CombinedFeedNarrow(),
+          messages: [message],
+          subscriptions: [eg.subscription(channel)],
+          navObservers: [navObserver],
+        );
+
+        assert(pushedRoutes.length == 1);
+        pushedRoutes.clear();
+
+        connection.prepare(json: eg.newestGetMessagesResult(
+          messages: [message],
+          foundOldest: true,
+        ).toJson());
```

Comment:
nit: use 2 lines for this instead of 4:
```suggestion
        connection.prepare(json: eg.newestGetMessagesResult(
          foundOldest: true, messages: [message]).toJson());
```

(and also put the arguments in the same order found at the declaration and other call sites)